gui/conversation.py

```diff
@@ -52,13 +69,42 @@ def keyPressEvent(self, event):
             # Let the parent class handle all other key events
             super().keyPressEvent(event)
 
-    def insertFromMimeData(self, mimeData):
-        if mimeData.hasText():
+    def insertFromMimeData(self, mimeData: QMimeData):
+        IMAGE_FORMATS = ('.png', '.jpg', '.jpeg', '.gif', '.webp')
+        if mimeData.hasImage():
+            image = QImage(mimeData.imageData())
+            if not image.isNull():
+                logger.debug("Inserting image from clipboard...")
+                temp_dir = tempfile.gettempdir()
+                mime_file_name = self.generate_unique_filename("image.png")
+                temp_file_path = os.path.join(temp_dir, mime_file_name)
+                image.save(temp_file_path)
+                self.add_image_thumbnail(image, temp_file_path)
+                self.main_window.add_image_to_selected_thread(temp_file_path)
+        elif mimeData.hasUrls():
+            logger.debug("Inserting image from URL...")
+            for url in mimeData.urls():
+                if url.isLocalFile():
+                    file_path = url.toLocalFile()
+                    logger.debug(f"Local file path: {file_path}")
+                    if file_path.lower().endswith(IMAGE_FORMATS):
+                        image = QImage(file_path)
+                        if not image.isNull():
+                            self.add_image_thumbnail(image, file_path)
+                            self.main_window.add_image_to_selected_thread(file_path)
+                        else:
+                            logger.error(f"Could not load image from file: {file_path}")
+                    else:
+                        logger.warning(f"Unsupported file type: {file_path}")
+                        QMessageBox.warning(self, "Error", "Unsupported file type. Please only upload image files.")
+                else:
+                    logger.warning(f"Non-local file URLs are not supported: {url.toString()}")
```

Comment:
Here we can rely on super method. This enables links on Mac
super().insertFromMimeData(mimeData)